modules/dnn/src/tflite/tflite_importer.cpp

```diff
@@ -764,6 +766,43 @@ void TFLiteImporter::parseTranspose(const Operator& op, const std::string& opcod
     addLayer(layerParams, op);
 }
 
+void TFLiteImporter::parseGlobalPooling(const Operator& op, const std::string& opcode, LayerParams& layerParams)
+{
+    layerParams.type = "Pooling";
+    if(!opcode.compare("MEAN")) {
+        layerParams.set("pool", "ave");
+    }
+    else if (!opcode.compare("REDUCE_MAX")) {
+        layerParams.set("pool", "max");
+    }
+    else {
+        CV_Error(Error::StsNotImplemented, "Unsupported pooling " + opcode);
+    }
+    layerParams.set("global_pooling", true);
+    auto options = op.builtin_options_as_ReducerOptions();
+    bool keep_dims = options->keep_dims();
+
+    if (!keep_dims) {
+        const std::string name = layerParams.name;
+        layerParams.name = layerParams.name + "_pool2d";
+        addLayer(layerParams, op);
+        LayerParams permFL;
+        permFL.set("axis", 1);
+        permFL.set("end_axis", 3);
+        int FLId = dstNet.addLayerToPrev(name, "Flatten", CV_32F, permFL);
```

Comment:
addLayerToPrev connects to the latest added layer. However, reduction may appear in a new branch during import. So get related input by `auto inpId = layerIds[op.inputs()->Get(0)];`.